File: JS/floating_number_format.js
function format_float(number_str){
    number_str = String(number_str);
    let trailing_index = number_str.indexOf('.');
    let trailing = '.00';
    if(trailing_index > -1){
        trailing = number_str.slice(trailing_index);
        number_str = number_str.slice(0, trailing_index);
    }

    let indices = [];
    let lt = number_str.length;
    let fin_str = '';
    if(lt > 3){
        let end = false;
        let e_index = lt;
        while(!end){
            let tmp_str = number_str.slice(0, e_index);
            let tmp_index = mini_format(tmp_str);
            indices.push(tmp_index);
            let fin_str = tmp_str.slice(0, tmp_index);
            if(fin_str.length > 3){
                e_index = tmp_index;
            }
            else {
                end = true;
            }
        }
    }

    indices.reverse();
    let start_index = 0;
    for(let i=0; i<=indices.length; i++){
        fin_str += number_str.slice(start_index, indices[i]);
        start_index = indices[i];
        if(i !== indices.length){
            fin_str += ',';
        }
    }

    return fin_str += trailing;

    function mini_format(num_str){
        let m_lt = num_str.length;
        return m_lt-3;
    }
}

module.exports = format_float;

File: Dart/floating_number_format.dart
String main() {
  String numStr = '0012033456.954';
  int trailingIndex = numStr.indexOf('.');
  String trailing = '.00';
  if (trailingIndex > -1) {
    trailing = numStr.substring(trailingIndex);
    if (trailing.length == 2) {
      trailing += '0';
    }
    if (trailing.length == 1) {
      trailing += '00';
    }
    numStr = numStr.substring(0, trailingIndex);
  }

  numStr = stripLeading(numStr);
  String fStr = '';
  int lt = numStr.length;
  int count = lt ~/ 3;
  List<String> strBits = [];

  for (int i = 0; i < count; i++) {
    String tmpStr = numStr.substring(numStr.length - 3);
    strBits.add(tmpStr);
    numStr = numStr.substring(0, numStr.length - 3);
  }

  if (numStr.length > 0) {
    strBits.add(numStr);
  }

  List<String> revStr = strBits.reversed.toList();

  revStr.forEach((element) {
    fStr += element;
    fStr += ',';
  });
  if (revStr.length > 0) {
    fStr = fStr.substring(0, fStr.length - 1);
  } else {
    fStr = '0';
  }
  fStr += trailing;
  print(fStr);
  return fStr;
}

String stripLeading(String numbStr) {
  if (numbStr.indexOf('0') == 0) {
    numbStr = numbStr.substring(1);
    if (numbStr.indexOf('0') == 0) {
      numbStr = stripLeading(numbStr);
    }
  }
  return numbStr;
}
